extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsTopologyManager.java

```diff
@@ -0,0 +1,167 @@
+package io.quarkus.kafka.streams.runtime;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.StreamSupport;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.ListTopicsResult;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.Topology;
+import org.eclipse.microprofile.config.Config;
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+/**
+ * Manages the lifecycle of a Kafka Streams pipeline. If there's a producer
+ * method returning a KS {@link Topology}, then this topology will be configured
+ * and started. Optionally, before starting the pipeline, this manager will wait
+ * for a given set of topics to be created, as KS itself will fail without all
+ * input topics being created upfront.
+ */
+@ApplicationScoped
+public class KafkaStreamsTopologyManager {
+
+    private static final String BOOTSTRAP_SERVERS_PROPERTY = "kafka.bootstrap.servers";
+    private static final String TOPIC_NAMES_PROPERTY = "quarkus.kafka.streams.topics";
+    private static final String STREAMS_OPTION_PREFIX = "quarkus.kafka.streams.";
```

Comment:
This should be handled by a proper config object.

For the options, you can use a `Map` and then push all the options.

I'm also wondering why the first one doesn't have a prefix? I suppose there's a reason but it's really an issue IMHO.

Reply:
Yes, `kafka.bootstrap.servers` has been defined by @cescoffier and also is supported by Reactive Messaging. I'm not sure whether he defined other "kafka..." options. In that light, should we perhaps simply use "kafka.streams." as a prefix? Might be better actually.